tests/countries/test_finland.py

```diff
@@ -165,6 +167,124 @@ def test_all_saints_day(self):
+        )
+        self.assertNoHolidayName(
+            name,
+            Finland(years=year_of_adoption - 1, categories=UNOFFICIAL),
+        )
+
+    def test_runeberg_day(self):
+        name = "Runebergin päivä"
+        self._test_unofficial_holiday(name, "02-05", 1854)
+
+    def test_minna_canth_day(self):
+        name = "Minna Canthin päivä, tasa-arvon päivä"
+        self._test_unofficial_holiday(name, "03-19", 2007)
+
+    def test_agricola_day(self):
+        name = "Mikael Agricolan päivä, suomen kielen päivä"
+        self._test_unofficial_holiday(name, "04-09", 1978)
+
+    def test_veterans_day(self):
+        name = "Kansallinen veteraanipäivä"
+        self._test_unofficial_holiday(name, "04-27", 1987)
+
+    def test_europe_day(self):
+        name = "Eurooppa-päivä"
+        self._test_unofficial_holiday(name, "05-09", 2019)
+
+    def test_finnish_identity_day(self):
+        self._test_unofficial_holiday("J.V. Snellmanin päivä, suomalaisuuden päivä", "05-12", 1952)
+
+    def test_remembrance_day(self):
+        self.assertHolidayName(
+            "Kaatuneitten muistopäivä",
+            self.unofficial_holidays,
+            "1977-05-15",
+            "1978-05-21",
+            "1985-05-19",
+            "2024-05-19",
+            "2025-05-18",
+        )
+        self.assertNoHoliday(
+            self.unofficial_holidays,
+            "1976-05-16",
+            "1975-05-18",
+        )
+
+    def test_eino_leino_day(self):
+        self._test_unofficial_holiday("Eino Leinon päivä, runon ja suven päivä", "07-06", 1992)
+
+    def test_finnish_nature_day(self):
+        self.assertHolidayName(
+            "Suomen luonnon päivä",
+            self.unofficial_holidays,
+            "2013-08-31",
+            "2014-08-30",
+            "2024-08-31",
+            "2025-08-30",
+            "2026-08-29",
+        )
+        self.assertNoHoliday(
+            self.unofficial_holidays,
+            "2012-08-25",
+            "2011-08-27",
+        )
+
+    def test_miina_sillanpaa_day(self):
+        self._test_unofficial_holiday(
+            "Miina Sillanpään ja kansalaisvaikuttamisen päivä", "10-01", 2016
+        )
+
+    def test_aleksis_kivi_day(self):
+        self._test_unofficial_holiday(
+            "Aleksis Kiven päivä, suomalaisen kirjallisuuden päivä", "10-10", 1950
+        )
+
+    def test_united_nations_day(self):
+        self._test_unofficial_holiday("YK:n päivä", "10-24", 1987)
+
+    def test_finnish_swedish_heritage(self):
+        self._test_unofficial_holiday(
+            "Ruotsalaisuuden päivä, Kustaa Aadolfin päivä", "11-06", 1908
+        )
+
+    def test_day_of_childrens_rights(self):
+        self._test_unofficial_holiday("Lapsen oikeuksien päivä", "11-20", 2020)
+
+    def test_jean_sibelius_day(self):
+        self._test_unofficial_holiday(
+            "Jean Sibeliuksen päivä, suomalaisen musiikin päivä", "12-08", 2007
+        )
+
```

Comment:
We need tests for newly added holidays (Kalevala Day etc). :)

Reply:
Ah, right 😅 Will add in a moment.

Reply:
Added!